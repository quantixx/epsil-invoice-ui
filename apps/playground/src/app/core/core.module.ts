//

import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { NavbarComponent } from './navbar/navbar.component';
import { PlaygroundAppRootComponent } from './playground-app-root/playground-app-root.component';

//

@NgModule({
  declarations: [
    PlaygroundAppRootComponent,
    PlaygroundAppRootComponent,
    NavbarComponent
  ],
  exports: [PlaygroundAppRootComponent],
  imports: [CommonModule, RouterModule]
})
export class CoreModule {}

export { PlaygroundAppRootComponent };
