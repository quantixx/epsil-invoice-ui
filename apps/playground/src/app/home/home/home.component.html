<!--  -->

<div class="jumbotron p-3 p-md-5 text-white rounded bg-dark">
  <div class="container">
    <h1 class="display-4">
      Welcome to the quantixx/epsil-invoice-ui playground
    </h1>
  </div>
</div>

<!--  -->

<div class="container">

  <p>Use the menu to navbar to go one of the mocked pages</p>

  <h2>General info</h2>

  <h3>Current
    <code>/invoices</code> response</h3>

  <pre>{{ data | async | json }}</pre>

</div>

<!--  -->
