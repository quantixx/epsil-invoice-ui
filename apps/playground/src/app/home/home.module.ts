//

import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { HomeComponent } from './home/home.component';

//

const routes: Routes = [{ path: '', component: HomeComponent }];

@NgModule({
  declarations: [HomeComponent],
  imports: [CommonModule, HttpClientModule, RouterModule.forChild(routes)]
})
export class HomeModule {}
