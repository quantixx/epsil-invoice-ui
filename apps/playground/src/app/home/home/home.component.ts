//

import { HttpClient } from '@angular/common/http';
import { Component, OnInit } from '@angular/core';

//

@Component({
  selector: 'epsil-playground-home',
  styleUrls: ['./home.component.scss'],
  templateUrl: './home.component.html'
})
export class HomeComponent implements OnInit {
  public data: object;

  constructor(private http: HttpClient) {}

  public ngOnInit() {
    this.data = this.http.get('http://localhost:5555/invoices');
  }
}
