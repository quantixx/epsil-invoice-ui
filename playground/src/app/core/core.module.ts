import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { PlaygroundAppRootComponent } from './playground-app-root/playground-app-root.component';

@NgModule({
  imports: [CommonModule, RouterModule],
  declarations: [PlaygroundAppRootComponent, PlaygroundAppRootComponent],
  exports: [PlaygroundAppRootComponent]
})
export class CoreModule {}

export { PlaygroundAppRootComponent };
