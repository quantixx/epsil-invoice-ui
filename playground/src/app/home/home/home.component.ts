import { Component, OnInit } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'my-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss']
})
export class HomeComponent implements OnInit {
  data: object;

  constructor(private http: HttpClient) {}

  ngOnInit() {
    this.data = this.http.get('http://localhost:5555/invoices');
  }
}
