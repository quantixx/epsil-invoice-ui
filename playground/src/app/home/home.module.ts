import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Routes, RouterModule } from '@angular/router';
import { HttpClientModule } from '@angular/common/http';

import { HomeComponent } from './home/home.component';

const routes: Routes = [{ path: '', component: HomeComponent }];

@NgModule({
  imports: [CommonModule, HttpClientModule, RouterModule.forChild(routes)],
  declarations: [HomeComponent]
})
export class HomeModule {}
