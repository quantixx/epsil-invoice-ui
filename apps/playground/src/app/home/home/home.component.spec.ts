//

import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { HomeComponent } from './home.component';

//

describe('Integration::NavbarComponent', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HomeComponent],
      imports: [RouterTestingModule]
    }).compileComponents();
  });

  it('should create', () => {
    const fixture = TestBed.createComponent(HomeComponent);

    const component = fixture.componentInstance;

    fixture.detectChanges();

    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
