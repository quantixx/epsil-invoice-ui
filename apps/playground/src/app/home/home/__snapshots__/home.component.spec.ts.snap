// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration::NavbarComponent should create 1`] = `
<epsil-playground-home
  data={[Function Observable]}
  http={[Function HttpClient]}
>
  
  
  <div
    class="jumbotron p-3 p-md-5 text-white rounded bg-dark"
  >
    
    
    <div
      class="container"
    >
      
      
      <h1
        class="display-4"
      >
        
        Welcome to the quantixx/epsil-invoice-ui playground
      
      </h1>
      
    
    </div>
    
  
  </div>
  
  
  
  <div
    class="container"
  >
    
  
    
    <p>
      Use the menu to navbar to go one of the mocked pages
    </p>
    
  
    
    <h2>
      General info
    </h2>
    
  
    
    <h3>
      Current
      
      <code>
        /invoices
      </code>
       response
    </h3>
    
  
    
    <pre>
      null
    </pre>
    
  
  
  </div>
  
  
  
</epsil-playground-home>
`;
